src/semantic-router/pkg/extproc/reason_mode_selector.go

```diff
@@ -32,16 +32,24 @@ func (r *OpenAIRouter) getReasoningModeAndCategory(query string) (bool, string)
 	// Normalize category name for consistent lookup
 	normalizedCategory := strings.ToLower(strings.TrimSpace(categoryName))
 
-	// Look up the category in the configuration
+	// Look up the category in the configuration and get the best model for it
 	for _, category := range r.Config.Categories {
 		if strings.EqualFold(category.Name, normalizedCategory) {
-			reasoningStatus := "DISABLED"
-			if category.UseReasoning {
-				reasoningStatus = "ENABLED"
+			// Get the best model for this category (first in the list)
```

Comment:
how to define a best model in category? if it is based on score, maybe we should sort the list and get the biggest one? instead of reading the first index